internal/go_repository_test.go

```diff
@@ -161,6 +167,25 @@ func TestModcacheRW(t *testing.T) {
 	}
 }
 
+func TestRepoCacheContainsGoEnv(t *testing.T) {
+	if err := bazel_testing.RunBazel("query", "@errors_go_mod//:go_default_library"); err != nil {
+		t.Fatal(err)
+	}
+	outputBase, err := getBazelOutputBase()
+	if err != nil {
+		t.Fatal(err)
+	}
+	goRoot := filepath.Join(outputBase, "external/go_sdk")
+	goPath := filepath.Join(outputBase, "external/bazel_gazelle_go_repository_cache")
+	goCache := filepath.Join(outputBase, "external/bazel_gazelle_go_repository_cache/gocache")
+	testtools.CheckFiles(t, goPath, []testtools.FileSpec{
+		{
+			Path:    "go.env",
+			Content: fmt.Sprintf("GOROOT='%s'\nGOPATH='%s'\nGOCACHE='%s'\nGOPRIVATE='example.com/m'\nGOSUMDB='off'", goRoot, goPath, goCache),
```

Comment:
This might be a little brittle. Instead, let's read the file and check whether it contains `GOPRIVATE='example.com/m'`.

Reply:
done.